### DISCUSSION TITLE
conda-forge/win-64/qt-5.12.9-h556501e_5.tar.bz2 doesn't show up in repodata

### DISCUSSION THREAD

**Discussion by isuruf** [conda/infra#514] [General]
Similar to https://github.com/conda/infra/discussions/11

**Comment by isuruf** [comment]
No build number 5 at
```bash
$ curl -sL https://conda.anaconda.org/conda-forge/win-64/repodata.json | grep qt-5.12.9
    "qt-5.12.9-h5909a2a_2.tar.bz2": {
    "qt-5.12.9-h5909a2a_3.tar.bz2": {
    "qt-5.12.9-h5909a2a_4.tar.bz2": {
    "qt-5.12.9-hb2cf2c5_0.tar.bz2": {
```

**Comment by jakirkham** [comment]
Thoughts @chenghlee @jezdez? 🙂

**Reply by chenghlee** [reply]
Likely the sync worker being grumpy about something; I'll take a look once I get off my calls.

**Comment by chenghlee** [comment]
The package artifact (https://anaconda.org/conda-forge/qt/5.12.9/download/win-64/qt-5.12.9-h556501e_5.tar.bz2) appears to be corrupt.

```
$ wget -P win-64 "https://anaconda.org/conda-forge/qt/5.12.9/download/win-64/qt-5.12.9-h556501e_5.tar.bz2"
$ sha256sum win-64/qt-5.12.9-h556501e_5.tar.bz2
2ccc1a3646e69e3541a36e41b50eee98f9ef27f7c3e4483432aec8ac6ff3c5eb  win-64/qt-5.12.9-h556501e_5.tar.bz2
$ tar tf win-64/qt-5.12.9-h556501e_5.tar.bz2 1>/dev/null

bzip2: Data integrity error when decompressing.
	Input file = (stdin), output file = (stdout)

It is possible that the compressed file(s) have become corrupted.
You can use the -tvv option to test integrity of such files.

You can use the `bzip2recover' program to attempt to recover
data from undamaged sections of corrupted files.

gtar: Unexpected EOF in archive
gtar: Error is not recoverable: exiting now
```

This is causing `conda index` to skip over the package and excluding it from `repodata.json`.

**Comment by izahn** [comment]
@Tobias-Fischer can you take a look at this when you get a chance? Thanks!

**Comment by Tobias-Fischer** [comment]
I'm not sure what's going on. When trying to rebuild, I get
```
conda.common.configuration.CustomValidationError: Parameter _channel_alias = '${TEST}' declared in <<merged>> is invalid.
channel_alias value '${TEST}' must have scheme/protocol.
```
straight away.

**Reply by izahn** [reply]
Hmm that is very strange! Will it help if I build it?

**Reply by Tobias-Fischer** [reply]
That'd be amazing, thanks!

**Reply by izahn** [reply]
Well I gave it a try but ran out of disk space on my little windows machine. Will try again later.

**Reply by izahn** [reply]
It didn't go well, ended up with
```
fatal error C1083: Cannot open include file: 'atlbase.h': No such file or directory
```
Anybody else able to give it a try?

**Comment by isuruf** [comment]
@chenghlee here's a link to a new package, but I can't seem to upload it myself.
https://drive.google.com/file/d/1cZFnYXvC5w4VywgiJkmySaPXDQUDvKIy/view?usp=sharing

**Reply by izahn** [reply]
Any chance we can get this uploaded? https://github.com/conda-forge/qtwebkit-feedstock/pull/21 is held up because of this.

**Reply by izahn** [reply]
Sorry to bother @conda-forge/core can someone please upload this?

**Reply by isuruf** [reply]
core cannot do anything. We have to wait for someone from Anaconda.

**Reply by chenghlee** [reply]
The updated build 6 is now available on the CDN:
```
$ conda search --platform=win-64 --override-channels -c https://conda-web.anaconda.org/conda-forge qt=5.12.9=h556501e_6
Loading channels: done
# Name                       Version           Build  Channel
qt                            5.12.9      h556501e_6  conda-forge
$ conda search --platform=win-64 --override-channels -c https://conda-static.anaconda.org/conda-forge qt=5.12.9=h556501e_6
Loading channels: done
# Name                       Version           Build  Channel
qt                            5.12.9      h556501e_6  conda-forge
```

**Reply by isuruf** [reply]
Thanks @chenghlee

**Reply by barabo** [reply]
Just for the record - *any* conda-forge core maintainer should be able to upload the file: `anaconda upload -u conda-forge qt-5.12.9-h556501e_6.tar.bz2`

Thanks @chenghlee for the quick assist!

**Reply by isuruf** [reply]
Not really. I tried uploading it to `cf-staging` and `isuruf` channel and they both became ghost files. See https://anaconda.org/cf-staging/qt and https://anaconda.org/isuruf/qt

**Reply by jakirkham** [reply]
Yeah for context @barabo we upload to `cf-staging` and then copy over to `conda-forge` after validating the package.

cc @beckermr (who knows this process well)

**Reply by beckermr** [reply]
I believe this is a bug/issue within anaconda.org as @isuruf said. Manual uploads do not use the staging process.

**Reply by jakirkham** [reply]
Sorry for confusing things then

**Reply by beckermr** [reply]
No worries and thanks for the bump!

**Comment by dholth** [comment]
`conda-forge/win-64/qt-5.12.9-h556501e_5.tar.bz2` file is corrupt and cannot be extracted all the way with `bzip2`. It should be added to the `removed` list in patches. I don't understand why it was able to be uploaded in the first place or where/whether we're getting partial metadata in the anaconda.org server - for example if you try to extract it with tar, and ignore the errors, you will see at least part of the info/ directory.

**Reply by beckermr** [reply]
Submitting a pr to admin requests on conda forge to mark it broken should do the trick.

**Reply by jakirkham** [reply]
https://github.com/conda-forge/admin-requests#mark-packages-as-broken-on-conda-forge
